chain-abci/src/staking/table.rs

```diff
@@ -317,12 +324,15 @@ impl StakingTable {
     pub(crate) fn insert_validator(&mut self, staking: &StakedState) {
         let val_addr = TendermintValidatorAddress::from(
             // no panic: Call ensure validator record exists.
-            &staking
-                .validator
-                .as_ref()
-                .unwrap()
-                .council_node
-                .consensus_pubkey,
+            match &staking.node_meta {
+                Some(NodeState::CouncilNode(v)) => v.council_node.consensus_pubkey.clone(),
+                _ => {
+                    // FIXME: the comment why it won't panic says "Call ensure validator record exists"
+                    // does it mean it's the caller's responsibility to ensure?
+                    // should the top-level comment have `# Panic` section, as this could panic if mis-used?
```

Comment:
yes, make better to let it return `Result`?

Reply:
returning result now where error signifies the caller didn't do the necessary validation